### PR TITLE
paddle.init() default use env

### PR DESCRIPTION
**Pull request by typhoonzero**
`paddle.init()` will use "PADDLE_*" environment variables to do the init. This will let cluster training jobs read arguments from environment variables which is provided by the cluster nodes.

### CODE REVIEW

File: python/paddle/v2/__init__.py [hunk 1]
```diff
@@ -42,6 +43,12 @@
 
 def init(**kwargs):
     args = []
+    # NOTE: append arguments if they are in ENV
+    for ek, ev in os.environ.iteritems():
+        if ek.startswith("PADDLE_"):
+            args.append('--%s=%s' % (ek.replace("PADDLE_", "").lower(), str(ev)))
```
Comment (Yancey1989) [inline python/paddle/v2/__init__.py]: `ek.replace("PADDLE_", "")` => `ek.lstrip("PADDLE_")` ?

Comment (typhoonzero) [inline python/paddle/v2/__init__.py]: lstrip will see `"PADDLE_"` as a iterative of chars, and remove the chars not string.

File: python/paddle/v2/__init__.py [hunk 2]
```diff
@@ -42,6 +43,13 @@
 
 def init(**kwargs):
     args = []
+    # NOTE: append arguments if they are in ENV
+    for ek, ev in os.environ.iteritems():
+        if ek.startswith("PADDLE_"):
+            args.append('--%s=%s' % (ek.replace("PADDLE_", "").lower(),
```
Comment (jacquesqiao) [inline python/paddle/v2/__init__.py]: 需不需要考虑重复配置的问题，比如代码也写了，然后环境变量也存在？

Comment (typhoonzero) [inline python/paddle/v2/__init__.py]: > overwrite arguments from ENV if it is in kwargs

如果paddle.init()传了参数会覆盖环境变量的配置。

Comment (typhoonzero) [inline python/paddle/v2/__init__.py]: 解决覆盖的问题。

**Comment by typhoonzero** [comment]
TODO: add documents for environments at: https://github.com/PaddlePaddle/Paddle/issues/1942

**Comment by jacquesqiao** [comment]
LGTM

**Pull request merged**